src/modules/emote_menu/stores/twitch-emotes.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable import/prefer-default-export */
+import twitchApi from '../../../utils/twitch-api.js';
+import Emote from '../../emotes/emote.js';
+import Icons from '../components/Icons.jsx';
+import debug from '../../../utils/debug.js';
+import {getEmoteFromRegEx} from '../../../utils/regex.js';
+
+const EMOTE_SET_QUERY = `
+query UserEmotes {
+  currentUser {
+    emoteSets {
+      emotes {
+        id,
+        token
+      },
+      id,
+      owner {
+        id,
+        displayName
+        profileImageURL(width: 300)
+      }
+    }
+  }
+}
+`;
+
+const TWITCH_EMOTE_CDN = (id, size) => `https://static-cdn.jtvnw.net/emoticons/v2/${id}/default/dark/${size}`;
+
+function getForcedProviderToChannels(key) {
+  switch (key) {
+    case '0':
+      return {
+        id: 'twitch-global',
+        displayName: 'Twitch Global',
+        icon: Icons.TWITCH,
+      };
+    case '33':
+      return {
+        id: 'twitch-turbo',
+        displayName: 'Twitch Turbo',
+        icon: Icons.PEOPLE,
+      };
+    case '42':
+      return {
+        id: 'twitch-turbo',
+        displayName: 'Twitch Turbo',
+        icon: Icons.PEOPLE,
+      };
+    case '457':
+      return {
+        id: 'twitch-turbo',
+        displayName: 'Twitch Turbo',
+        icon: Icons.PEOPLE,
+      };
+    case '793':
+      return {
+        id: 'twitch-turbo',
+        displayName: 'Twitch Turbo',
+        icon: Icons.PEOPLE,
+      };
+    case '19151':
+      return {
+        id: 'twitch-gaming',
+        displayName: 'Twitch Gaming',
+        icon: Icons.TWITCH_GAMING,
+      };
+    case '19194':
+      return {
+        id: 'twitch-gaming',
+        displayName: 'Twitch Gaming',
+        icon: Icons.TWITCH_GAMING,
+      };
+    default:
+      return {
+        id: -1,
+        displayName: 'Unlocked',
+        icon: Icons.UNLOCK,
+      };
+  }
+}
+
+export async function loadTwitchEmotes() {
+  try {
```

Comment:
why is this entire operation wrapped in a try catch? wouldn't it just be the http call?